barretenberg/cpp/src/barretenberg/relations/utils.hpp

```diff
@@ -58,14 +63,36 @@ template <typename Flavor> class RelationUtils {
         apply_to_tuple_of_tuples(tuple, set_to_zero);
     }
 
+    /**
+     * @brief Scale Univaraites, each representing a subrelation, by different challenges
+     *
+     * @param tuple Tuple of tuples of Univariates
+     * @param challenge Array of NUM_SUBRELATIONS - 1 challenges (because the first subrelation doesn't need to be
+     * scaled)
+     * @param current_scalar power of the challenge
+     */
+    static void scale_univariates(auto& tuple, const AlphaType& challenges, FF& current_scalar)
+        requires proof_system::IsFoldingFlavor<Flavor>
+    {
+        size_t idx = 0;
+        std::array<FF, NUM_SUBRELATIONS> tmp{ current_scalar };
+        std::copy(challenges.begin(), challenges.end(), tmp.begin() + 1);
+        auto scale_by_challenges = [&]<size_t, size_t>(auto& element) {
+            element *= tmp[idx];
+            idx++;
+        };
+        apply_to_tuple_of_tuples(tuple, scale_by_challenges);
+    }
+
     /**
      * @brief Scale Univaraites by consecutive powers of the provided challenge
      *
      * @param tuple Tuple of tuples of Univariates
      * @param challenge
      * @param current_scalar power of the challenge
      */
-    static void scale_univariates(auto& tuple, const FF& challenge, FF& current_scalar)
+    static void scale_univariates(auto& tuple, const AlphaType& challenge, FF& current_scalar)
```

Comment:
I initially wanted to take that approach and started implementing it it. However, because ECCVM, Translator and AVM have many subrelations (from discussing with Kesha I gathered that AVM might have hundreads of subrelations) a unified approach didn't seem a great idea. This is why I've done all the maneuvering here. Maybe it's a bit awkward and can be improved. Definitely open to that.